src/components/network/network-loader.js

```diff
@@ -19,117 +19,293 @@ genotet.utils.inherit(genotet.NetworkLoader, genotet.ViewLoader);
 /**
  * Loads the network data, adding the genes given by geneRegex.
  * @param {string} fileName Network File name.
- * @param {string} geneRegex Regex for gene selection.
+ * @param {string} inputGenes The input genes.
+ * @param {boolean} isRegex Whether inputGene is regex.
  * @override
  */
-genotet.NetworkLoader.prototype.load = function(fileName, geneRegex) {
-  this.loadNetwork_(fileName, geneRegex);
+genotet.NetworkLoader.prototype.load = function(fileName, inputGenes, isRegex) {
+  var genes = this.prepareGene_(inputGenes, isRegex);
+  this.data.genes = genes;
+  this.loadNetwork_(fileName, genes);
+};
+
+/**
+ * Prepares gene array for the input gene string and isRegex.
+ * @param {string} inputGenes Input string for genes.
+ * @param {boolean} isRegex If inputGene is regex.
+ * @return {!Array<string>}
+ * @private
+ */
+genotet.NetworkLoader.prototype.prepareGene_ = function(inputGenes, isRegex) {
+  var genes = [];
+  if (isRegex) {
+    var regex;
+    try {
+      regex = RegExp(inputGenes, 'i');
+    } catch (e) {
+      genotet.error('invalid gene regex', inputGenes);
+      return [];
+    }
+    this.data.networkInfo.nodes.forEach(function(node) {
```

Comment:
Because genes is an array just containing the string, or node.id, not the node object.